src/Fixer/ClassNotation/ClassDefinitionFixer.php

```diff
@@ -362,15 +363,37 @@ private function makeClassyDefinitionSingleLine(Tokens $tokens, int $startIndex,
     {
         for ($i = $endIndex; $i >= $startIndex; --$i) {
             if ($tokens[$i]->isWhitespace()) {
-                if ($tokens[$i - 1]->isComment() || $tokens[$i + 1]->isComment()) {
+                if (str_contains($tokens[$i]->getContent(), "\n")) {
+                    if (\defined('T_ATTRIBUTE')) { // @TODO: drop condition and else when PHP 8.0+ is required
+                        if ($tokens[$i - 1]->isGivenKind(CT::T_ATTRIBUTE_CLOSE) || $tokens[$i + 1]->isGivenKind(T_ATTRIBUTE)) {
+                            continue;
```

Comment:
Can we extract some logic to separate methods? This is 4th level `if`, and all 4 are already in `for` loop.

Reply:
The last level if call `continue` of the top loop, thus not easily.